src/Microsoft.Diagnostics.Runtime/ClrHeap.cs

```diff
@@ -213,6 +213,12 @@ virtual public ClrType GetTypeByMethodTable(ulong methodTable)
         /// <returns>An enumerator for all objects on the heap.</returns>
         abstract public IEnumerable<ulong> EnumerateObjectAddresses();
 
+        /// <summary>
+        /// Enumerates all objects on the heap and for each returns its address, type and size in an optimized way.
+        /// </summary>
+        /// <returns>An enumerator for all objects on the heap.</returns>
+        abstract public IEnumerable<Tuple<ulong, ulong, ClrType, ulong>> EnumerateObjectDetails();
```

Comment:
We should return a specific, documented type instead of a tuple.

Reply:
You're right: I did not know where to define it...